src/orderbook.ts

```diff
@@ -9,8 +9,44 @@ export class Price {
     this.denominator = denominator;
   }
```

Comment:
Also, I just remembered, TS has this cool feature:
```ts
export class Fraction {
    constructor(
        public readonly numerator: number,
        public readonly denominator: number,
    ) {
        if (denominator === 0) {
            throw new Error("error");
        }
    }
}
```

This will make both the properties read-only and automagically set their value from the constructor arguments.

Reply:
This is cool. I wanted to allow instantiation with number as well as with BN, so I had to create some matching logic in the constructor.